### PR TITLE
Add functionality to specify batch name

### PR DESCRIPTION
**Pull request by negar-rostamzadeh**

### CODE REVIEW

File: smartdispatch/utils.py [hunk 1]
```diff
@@ -27,6 +27,14 @@ def jobname_generator(jobname, job_id):
         croped_string = '{}_{}'.format(jobname, job_id)
     return croped_string
 
+def default_name(jobname, job_generator):
```
Comment (mgermain) [inline smartdispatch/utils.py]: I don't get that .... this is all already handled by your previous PR with the function `jobname_generator`, you don't need that.

File: scripts/smart-dispatch [hunk 2]
```diff
@@ -129,6 +131,7 @@ def main():
 
     job_generator = job_generator_factory(queue, commands, command_params, CLUSTER_NAME, path_job)
     # generating default names per each jobs in each batch
+    jobname = utils.default_name(jobname, job_generator)
```
Comment (mgermain) [inline scripts/smart-dispatch]: You don't need this. It's done 2 line lower at line 136.

File: scripts/smart-dispatch [hunk 3]
```diff
@@ -166,6 +169,7 @@ def main():
 def parse_arguments():
     parser = argparse.ArgumentParser()
     parser.add_argument('-q', '--queueName', required=True, help='Queue used (ex: qwork@mp2, qfat256@mp2, qfat512@mp2)')
+    parser.add_argument('-N', '--batchName', required=False, help='The names of batches')
```
Comment (mgermain) [inline scripts/smart-dispatch]: `help='The batch name. Max 235 chars.'`

Also, add a check for the max length. Like the other check lower in the function.

File: smartdispatch/utils.py (hunk 1 above)
Comment (mgermain) [inline smartdispatch/utils.py]: There is still an extra space missing

File: scripts/smart-dispatch (hunk 3 above)
Comment (mgermain) [inline scripts/smart-dispatch]: You didn't add the check :(

File: scripts/smart-dispatch [hunk 4]
```diff
@@ -164,8 +166,12 @@ def main():
 
 
 def parse_arguments():
+    # This is caused by filesystem limitations.
+    batchNameMaxLen = 235
+    
     parser = argparse.ArgumentParser()
-    parser.add_argument('-q', '--queueName', required=True, help='Queue used (ex: qwork@mp2, qfat256@mp2, qfat512@mp2)')
+    parser.add_argument('-q', '--queueName', required=True, help='Queue used (ex: qwork@mp2, qfat256@mp2, gpu_1)')
```
Comment (MarcCote) [inline scripts/smart-dispatch]: What is `gpu_1`? Was it plan to remove `qfat512@mp2`.

File: scripts/smart-dispatch [hunk 5]
```diff
@@ -164,8 +166,12 @@ def main():
 
 
 def parse_arguments():
+    # This is caused by filesystem limitations.
+    batchNameMaxLen = 235
```
Comment (MarcCote) [inline scripts/smart-dispatch]: Python naming convention, for variables, is to use lowercase and an underscore to separate words.

**Comment by MarcCote** [comment]
On a side note, this PR has no description and an unclear title. What is Iss02?

**Comment by mgermain** [comment]
fixes #141 
fixes #147 
fixes #55

**Pull request merged**